srt-protocol/src/protocol/receiver/buffer.rs

```diff
@@ -218,92 +241,89 @@ impl ReceiveBuffer {
 
     // index in buffer for a given sequence number
     // returns None if before the start of the buffer. Might return a index out of bounds
-    fn index_for_seqno(&self, s: SeqNumber) -> Option<usize> {
-        if s < self.seqno0 {
+    fn index_for_seqno(&self, seq_number: SeqNumber) -> Option<usize> {
+        if seq_number < self.seqno0 {
             None
         } else {
-            Some(usize::try_from(s - self.seqno0).unwrap())
+            let index = usize::try_from(seq_number - self.seqno0).unwrap();
+            //            assert!(index < self.buffer.len(), "{:?} !< {:?}", index, self.buffer.len());
+            Some(index)
```

Comment:
This can probably be rolled back--I assume you found that that assert was not true?

Reply:
Yes, and I revisited the code, in a type system as powerful as Rust's is, I find defensive runtime assertions in production code to be a code smell, often times a result of Primitive Obsession.